hazelcast/src/test/java/com/hazelcast/map/impl/querycache/QueryCacheTest.java

```diff
@@ -230,6 +235,39 @@ public void getAll_with_non_existent_keys() {
         queryCache.getAll(keySet);
     }
 
+    @Test
+    public void testQueryCache_with_mapAttribute_inPredicate() {
+
+        String MAP_ATTRIBUTE_NAME = "booleanMapAttribute";
+
+        Config config = new Config();
+        config.getMapConfig(mapName)
+                .addQueryCacheConfig(
+                        new QueryCacheConfig(cacheName)
+                                .setIncludeValue(true)
+                                .setPredicateConfig(// use map attribute in a predicate
+                                        new PredicateConfig(Predicates.equal(MAP_ATTRIBUTE_NAME, true))
+                                ))
+                .addMapAttributeConfig(
+                        new MapAttributeConfig()
+                                .setExtractor(EvenNumberEmployeeValueExtractor.class.getName())
+                                .setName(MAP_ATTRIBUTE_NAME));
+
+        IMap<Integer, Employee> map = getIMap(config);
+        QueryCache<Integer, Employee> queryCache = map.getQueryCache(cacheName);
+
+        populateMap(map, 100);
+
+        assertEquals(queryCache.keySet().size(), 50);
```

Comment:
This should be `assertQueryCacheSizeEventually` as in other tests in this test class. Because queryCache population is an async process.

Reply:
Done.